tools/wake-format/formatter.h

```diff
@@ -59,6 +62,15 @@ struct ctx_t {
   }
 };
 
+struct token_traits_t {
+  std::vector<CSTElement> before_bound = {};
+  std::vector<CSTElement> after_bound = {};
+  void bind_before(CSTElement e) { before_bound.push_back(e); }
+  void bind_after(CSTElement e) { after_bound.push_back(e); }
```

Comment:
I think one issue with this interface is that you have to order your calls to them precisely and in an inconvenient way. Two options 1) use a stack for `before_bound` and a queue for `after_bound` (or use a deque for both) or 2) use an std::map with an ordering on the CSTElements based on their token order so that you can call them in any order you please